File: integrations/wasm/go/geolocation.go
package angularwasm

// GeolocationValue is the serializable snapshot written by the AngularTS
// geolocation directive.
type GeolocationValue struct {
	Latitude         float64  `json:"latitude"`
	Longitude        float64  `json:"longitude"`
	Accuracy         float64  `json:"accuracy"`
	Altitude         *float64 `json:"altitude"`
	AltitudeAccuracy *float64 `json:"altitudeAccuracy"`
	Heading          *float64 `json:"heading"`
	Speed            *float64 `json:"speed"`
	Timestamp        float64  `json:"timestamp"`
}

File: integrations/wasm/rust/crates/angular-ts/src/geolocation.rs
/// Serializable location snapshot written by the AngularTS `geolocation`
/// directive.
#[derive(Debug, Clone, Copy, PartialEq)]
#[cfg_attr(target_arch = "wasm32", derive(serde::Deserialize, serde::Serialize))]
#[cfg_attr(target_arch = "wasm32", serde(rename_all = "camelCase"))]
pub struct GeolocationValue {
    pub latitude: f64,
    pub longitude: f64,
    pub accuracy: f64,
    pub altitude: Option<f64>,
    pub altitude_accuracy: Option<f64>,
    pub heading: Option<f64>,
    pub speed: Option<f64>,
    pub timestamp: f64,
}
